fe/src/main/java/org/apache/doris/task/DeleteJob.java

```diff
@@ -0,0 +1,190 @@
+        this.label = label;
+        this.deleteInfo = deleteInfo;
+        totalTablets = Sets.newHashSet();
+        finishedTablets = Sets.newHashSet();
+        quorumTablets = Sets.newHashSet();
+        tabletDeleteInfoMap = Maps.newConcurrentMap();
+        pushTasks = Sets.newHashSet();
+        state = DeleteState.UN_QUORUM;
+    }
+
+    /**
+     * check and update if this job's state is QUORUM_FINISHED or FINISHED
+     * The meaning of state:
+     * QUORUM_FINISHED: For each tablet there are more than half of its replicas have been finished
+     * FINISHED: All replicas of this jobs have finished
+     */
+    public void checkAndUpdateQuorum() throws MetaNotFoundException {
+        long dbId = deleteInfo.getDbId();
+        long tableId = deleteInfo.getTableId();
+        long partitionId = deleteInfo.getPartitionId();
+        Database db = Catalog.getInstance().getDb(dbId);
+        if (db == null) {
+            throw new MetaNotFoundException("can not find database "+ dbId +" when commit delete");
+        }
+
+        short replicaNum = -1;
+        db.readLock();
+        try {
+            OlapTable table = (OlapTable) db.getTable(tableId);
+            if (table == null) {
+                throw new MetaNotFoundException("can not find table "+ tableId +" when commit delete");
+            }
+            replicaNum = table.getPartitionInfo().getReplicationNum(partitionId);
+        } finally {
+            db.readUnlock();
+        }
+
+        short quorumNum = (short) (replicaNum / 2 + 1);
+        for (TabletDeleteInfo tDeleteInfo : getTabletDeleteInfo()) {
+            if (tDeleteInfo.getFinishedReplicas().size() == replicaNum) {
+                finishedTablets.add(tDeleteInfo.getTabletId());
+            }
+            if (tDeleteInfo.getFinishedReplicas().size() >= quorumNum) {
+                quorumTablets.add(tDeleteInfo.getTabletId());
+            }
+        }
+        LOG.info("check delete job quorum, transaction id: {}, total tablets: {}, quorum tablets: {},",
+                signature, totalTablets.size(), quorumTablets.size());
+
+        if (finishedTablets.containsAll(totalTablets)) {
+            setState(DeleteState.FINISHED);
+        } else if (quorumTablets.containsAll(totalTablets)) {
+            setState(DeleteState.QUORUM_FINISHED);
+        }
+    }
+
+    public void setState(DeleteState state) {
+        this.state = state;
+    }
+
+    public DeleteState getState() {
+        return this.state;
+    }
+
+    public boolean addTablet(long tabletId) {
+        return totalTablets.add(tabletId);
+    }
+
+    public boolean addPushTask(PushTask pushTask) {
+        return pushTasks.add(pushTask);
+    }
+
+    public boolean addFinishedReplica(long tabletId, Replica replica) {
+        tabletDeleteInfoMap.putIfAbsent(tabletId, new TabletDeleteInfo(tabletId));
+        TabletDeleteInfo tDeleteInfo =  tabletDeleteInfoMap.get(tabletId);
+        synchronized (tDeleteInfo) {
```

Comment:
No need to use `synchronized` here, I think you can just use a `ConcurrentSet` in `TabletDeleteInfo`